### PR TITLE
Add support to the Operator for setting default labels on Keycloak pods

### LINKED ISSUE #20625: Add support to the Operator for setting default labels on Keycloak pods
### Description

The Operator needs to be able to set some default set of labels on Keycloak pods. Those labels will be defined directly in the Operator code.

Same functionality in the old Operator: https://github.com/keycloak/keycloak-operator/blob/4b9abfb2922643cd6eb847fa7dacbba6bd9e2c81/pkg/model/constants.go#L66

### Discussion

_No response_

### Motivation

_No response_

### Details

_No response_

### PR DESCRIPTION
**Pull request by vmuzikar**
Closes #20625

Also sneaked in one fix for our Kustomize file that is used for dev purposes only.

### CODE REVIEW

File: operator/src/main/resources/application-rhbk.properties [hunk 1]
```diff
@@ -0,0 +1,7 @@
+operator.keycloak.pod-labels."com.company"=Red_Hat
+operator.keycloak.pod-labels."rht.prod_name"=Red_Hat_Runtimes
+operator.keycloak.pod-labels."rht.prod_ver"=${SSO_QUARTER:sso_quarter_placeholder}
+operator.keycloak.pod-labels."rht.comp"=SSO
+operator.keycloak.pod-labels."rht.comp_ver"=${SSO_VER:${quarkus.application.version}}
+operator.keycloak.pod-labels."rht.subcomp"=SSO_Server
+operator.keycloak.pod-labels."rht.subcomp_t"=application
```
Comment (vmuzikar) [inline operator/src/main/resources/application-rhbk.properties:7]: @ASzc Could you please check it's ok for the prod environment? `SSO_QUARTER` and `SSO_VER` are env vars with some default values (`quarkus.application.version` probably is reasonable default and might not even need overriding).

Please also note you need to build the Operator with the `-Poperator-prod` profile for this file to be used.

Comment (ASzc) [reply operator/src/main/resources/application-rhbk.properties:7]: This is implemented differently to how the golang one did it, but I assume you've checked and the labels are appropriately applied at runtime.

Example values:

```
SSO_QUARTER: 2023_Q2
SSO_VER: 7.6.3
```

So those defaults seem reasonable. I will probably set them anyway in the CSV

And noted. I can swap the profile after this is merged

**Review by ASzc — DISMISSED** [review]
Seems ok for product

File: operator/pom.xml [hunk 2]
```diff
@@ -289,6 +289,13 @@
                         </goals>
                     </execution>
                 </executions>
+                <configuration>
+                    <systemPropertyVariables>
+                        <OPERATOR_TEST_LABEL_EXPRESSION>my-value</OPERATOR_TEST_LABEL_EXPRESSION>
```
Comment (mabartos) [inline operator/pom.xml:294]: It's out of the scope of this PR, but IMHO, the operator module contains a lot of different contexts related to the environment. I mean, we have at the same place dev, prod, test related stuff, and also the examples. IMO, it's starting to be a bit messy and could be more challenging to orient yourself to what's essential and what's not.

Maybe we could consider some follow-up tasks related to the separation of these contexts a little bit. @vmuzikar WDYT?

**Review by mabartos — DISMISSED** [review]
@vmuzikar LGTM. I've added only one comment to the code; out of the scope of this PR.

File: operator/pom.xml (hunk 2 above)
Comment (vmuzikar) [reply operator/pom.xml:295]: @mabartos Done.

File: operator/src/main/resources/application-rhbk.properties [hunk 3]
```diff
@@ -0,0 +1,7 @@
+operator.keycloak.pod-labels."com.company"=Red_Hat
+operator.keycloak.pod-labels."rht.prod_name"=Red_Hat_Runtimes
+operator.keycloak.pod-labels."rht.prod_ver"=${SSO_QUARTER:sso_quarter_placeholder}
```
Comment (mabartos) [inline operator/src/main/resources/application-rhbk.properties:3]: @vmuzikar Should also be updated, right?

Comment (mabartos) [reply operator/src/main/resources/application-rhbk.properties:3]: Explained in the private conversation.

**Review by mabartos — APPROVED** [review]
@vmuzikar LGTM, thanks!

File: operator/src/main/resources/application-rhbk.properties (hunk 3 above)
Comment (vmuzikar) [reply operator/src/main/resources/application-rhbk.properties:3]: Discussed offline, thanks

**Review by pedroigor — APPROVED** [review]

**Pull request merged**